### DISCUSSION TITLE
Axis tick marks

### DISCUSSION THREAD

**Discussion by mattjvincent** [genome-spy/genome-spy#81] [Q&A] [answered]
Is there a way to have the axis tick marks stretch across the plot?

Maybe this is the same as [Issue #52](https://github.com/genome-spy/genome-spy/issues/52).

I know about `rule` and that works for fixed data.  However, I want the `rule` to line up with the axis tick marks.  Is there a setting to do that? 

<img width="642" alt="axis" src="https://user-images.githubusercontent.com/1842565/151247078-8b925760-6b80-4756-b259-1545d16f2766.png">

**Comment by tuner** [comment]
Ah, you are asking for grid lines #58. Yes, a proper implementation depends on #52, but it also depends on some smaller-scale refactoring. Namely, the tick generator: https://github.com/genome-spy/genome-spy/blob/2abd5a2fb3e752bf651a7ff8fa4017df58fc12ab/packages/core/src/view/axisView.js#L151 ... should be extracted into a proper data source or transform that you (and GenomeSpy internally) could use to make custom grid lines.

So, the answer to your question is: not currently, but once I or someone else does the datasource-refactoring. However, there's a workaround that might or might not work in your case: you can specify the ticks values explicitly so that they align with your grid lines. Something like:

```js
{
  y: {
    ...,
    axis: {
      values: [0, 4, 8, 12]
    }
  }
}
```

**Comment by tuner** [comment, marked as answer]
Grid lines are nowadays supported, btw.

https://genomespy.app/docs/grammar/scale/#axes
